app/search_filter/search_filter.component.html

```diff
@@ -0,0 +1,6 @@
+<div *ngIf="selectedFilter">
+  <div *ngFor="let option of selectedFilter.options">
+    <input  type="checkbox" [checked]="option.selected" value="{{option.value}}"
+            (change)="optionSelected($event.target.value)" /> {{option.name}} <br>
```

Comment:
why not pass `option` here instead?
or even do `option.selected = !option.selected`?